src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs

```diff
@@ -25,9 +26,9 @@ public class ListDatabasesRequestDatabaseProperties
     /// </summary>
     static class ListDatabaseRequestHandlerFactory
     {
-        public static IListDatabaseRequestHandler getHandler(bool includeDetails, bool isSqlDB)
+        public static IListDatabaseRequestHandler getHandler(bool includeDetails, bool isSqlDB, ConnectionInfo connectionInfo = null)
         {
-            if (!includeDetails)
+            if (!includeDetails || (connectionInfo != null && connectionInfo.EngineEdition == DatabaseEngineEdition.SqlOnDemand))
```

Comment:
`connectionInfo?.EngineEdition == DatabaseEngineEdition.SqlOnDemand`

Reply:
@Charles-Gagnon - there is another STS change on the way. 
I'll be merging the changes so there is only one review.

The comment is resolved in the PR that is on the way.